sparkle/src/components/Table.tsx

```diff
@@ -0,0 +1,296 @@
+  return (
+    <table
+      className={classNames("s-w-full s-border-collapse", className || "")}
+      {...props}
+    >
+      {children}
+    </table>
+  );
+};
+
+const Header: React.FC<HeaderProps> = ({ children, className, ...props }) => {
+  return (
+    <thead
+      className={classNames("s-text-xs s-capitalize", className || "")}
+      {...props}
+    >
+      {children}
+    </thead>
+  );
+};
+
+const Head: React.FC<HeadProps> = ({
+  children,
+  className,
+  width,
+  ...props
+}) => {
+  return (
+    <th
+      className={classNames(
+        "s-py-1 s-pr-3 s-text-left s-font-medium s-text-element-800",
+        width === "expanded" ? "s-w-full" : "s-w-auto",
+        className || ""
+      )}
+      {...props}
+    >
+      <div className="s-flex s-items-center s-space-x-1 s-whitespace-nowrap">
+        {children}
+      </div>
+    </th>
+  );
+};
+
+const Body: React.FC<React.HTMLAttributes<HTMLTableSectionElement>> = ({
+  children,
+  className,
+  ...props
+}) => (
+  <tbody className={className || ""} {...props}>
+    {children}
+  </tbody>
+);
+
+const Footer: React.FC<React.HTMLAttributes<HTMLTableSectionElement>> = ({
+  children,
+  className,
+  ...props
+}) => (
+  <tfoot
+    className={classNames(
+      "s-border-t s-border-structure-200 s-bg-structure-50",
+      className || ""
+    )}
+    {...props}
+  >
+    {children}
+  </tfoot>
+);
+
+const Row: React.FC<RowProps> = ({
+  children,
+  className,
+  clickable = false,
+  onClick,
+  ...props
+}) => (
+  <tr
+    className={classNames(
+      "s-border-b s-border-structure-200 s-text-sm",
+      className || ""
+    )}
+    {...props}
+  >
+    {children}
+    {clickable && (
+      <td
+        className="s-w-1 s-cursor-pointer s-pl-1 s-text-element-600"
+        onClick={clickable ? onClick : undefined}
+      >
+        <Icon visual={MoreIcon} size="sm" />
+      </td>
+    )}
+  </tr>
+);
+
+const Cell: React.FC<CellProps> = ({
+  children,
+  className,
+  avatarUrl,
+  icon,
+  description,
+  ...props
+}) => (
+  <td
+    className={classNames(
+      "s-whitespace-nowrap s-py-2 s-pl-0.5 s-text-element-800",
+      className || ""
+    )}
+    {...props}
+  >
+    <div className="s-flex">
+      {avatarUrl && <Avatar visual={avatarUrl} size="xs" className="s-mr-2" />}
+      {icon && (
+        <Icon visual={icon} size="sm" className="s-mr-2 s-text-element-600" />
+      )}
+      <div className="s-flex">
+        <span className="s-text-sm s-text-element-800">{children}</span>
+        {description && (
+          <span className="s-pl-2 s-text-sm s-text-element-600">
+            {description}
+          </span>
+        )}
+      </div>
+    </div>
+  </td>
+);
+
+const Caption: React.FC<React.HTMLAttributes<HTMLTableCaptionElement>> = ({
+  children,
+  className,
+  ...props
+}) => (
+  <caption className={classNames(className || "")} {...props}>
+    {children}
+  </caption>
+);
+
+export const TableData = Object.assign(TableRoot, {
```

Comment:
We can avoid object assign and instead creates the sub components directly on the main component.